src/plugins/converter/converter.plugin.ts

```diff
@@ -0,0 +1,39 @@
+import { SpotterOption, SpotterPlugin, SpotterPluginLifecycle } from '../../core';
+import icon from './icon.png';
+export class ConverterPlugin extends SpotterPlugin implements SpotterPluginLifecycle {
+
+  identifier = 'Converter';
+
+  async onQuery(query: string): Promise<SpotterOption[]> {
+       
+    try {
+       const splitted = query.split(" ",4)
+       const convertFromAmount = Number(splitted[0])
+       const convertFromCurrencyType = splitted[1].toUpperCase()
+       const convertToCurrencyType = splitted[3].toUpperCase()
+       
+       if(splitted.length==4 && convertToCurrencyType.length==3)
+       {
+        var result=0
+        const data = await fetch(`https://api.exchangeratesapi.io/latest?base=${convertFromCurrencyType}&symbols=${convertToCurrencyType}`)
+        const json = data.json()
+        const r = json.then(data =>{
```

Comment:
Please use human understandable variables 🙂.

But here could be match simple
```ts
const data = await fetch(`https://api.exchangeratesapi.io/latest?base=${convertFromCurrencyType}&symbols=${convertToCurrencyType}`).then(d => d.json());
const rate = ...
```

Reply:
also, cache and reuse results. No need to query the server mutltiple times when a single request to download and store the conversion rate is enough.